### PR TITLE
key difficulty checks in task session

### LINKED ISSUE #2239: Add checks for key difficulty in task session
Currently key difficulty is only check in peersession, but nodes with lower key difficulty than expected can still compute tasks for each other. Add key difficulty check in task session as well.

### PR DESCRIPTION
**Pull request by tworec**
resolves #2239

### CODE REVIEW

File: tests/golem/network/p2p/test_peersession.py [hunk 1]
```diff
@@ -146,10 +145,10 @@ def test_handshake_server_protoid(self, send_mock):
...
                 reason=message.Disconnect.REASON.ProtocolVersion).slots())
 
-    @mock.patch('golem.network.transport.session.BasicSession.send')
-    def test_react_to_hello_key_not_difficult(self, send_mock):
-        self.peer_session.p2p_service.keys_auth.is_pubkey_difficult = \
-            mock.Mock(return_value=False)
+    @patch('golem.network.transport.session.BasicSession.send')
+    @patch('golem.core.keysauth.KeysAuth.is_pubkey_difficult')
+    def test_react_to_hello_key_not_difficult(self, is_difficult_fn, send_mock):
+        is_difficult_fn.return_value = False
```
Comment (etam) [inline tests/golem/network/p2p/test_peersession.py]: You can specify this at decorator, like this:
```python
@patch('golem.core.keysauth.KeysAuth.is_pubkey_difficult', return_value=False)
```

Comment (tworec) [reply tests/golem/network/p2p/test_peersession.py]: oooh, I was looking for this in docs, but given up. Thx! :)

File: tests/golem/task/test_tasksession.py [hunk 2]
```diff
@@ -236,29 +235,90 @@ def test_send_report_computed_task(self, get_mock):
...
+        difficulty = 4
+        conn = MagicMock()
+        ts = TaskSession(conn)
+        ts.task_server = Mock()
+        ts.task_server.config_desc = Mock()
+        ts.task_server.config_desc.key_difficulty = difficulty
+        ts.disconnect = Mock()
+        ts.send = Mock()
+
+        from golem.core.keysauth import KeysAuth
```
Comment (etam) [inline tests/golem/task/test_tasksession.py]: import at top

**Review by etam — CHANGES_REQUESTED** [review]

File: tests/golem/task/test_tasksession.py (hunk 2 above)
Comment (tworec) [reply tests/golem/task/test_tasksession.py]: ok

**Review by etam — APPROVED** [review]

**Pull request merged**